cmd/control/tlsconf.go

```diff
@@ -65,16 +65,21 @@ func writeCerts(generateServer bool, hostname []string, cfg *config.CloudConfig,
 			return err
 		}
 
-		return cfg.SetConfig(&config.CloudConfig{
-			Rancher: config.RancherConfig{
-				Docker: config.DockerConfig{
-					CAKey:      cfg.Rancher.Docker.CAKey,
-					CACert:     cfg.Rancher.Docker.CACert,
-					ServerCert: string(cert),
-					ServerKey:  string(key),
+		cfg, err = cfg.Merge(map[interface{}]interface{}{
```

Comment:
Why can't this be done in a type safe way?  Should we add a `MergeCloudConfig(config.CloudConfig)` method?  I just don't like all these hard coded strings.  And move then to const doesn't make it better.  I'd rather it be type safe.